### PR TITLE
Adding test for lfgv2

### PR DESCRIPTION
**Pull request by roquesrolando**

### CODE REVIEW

File: test/LFGFactoryV2.test.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+  const blocksPerDay = 42000;
+  const blocksPerWeek = blocksPerDay * 7;
+  const twoYearsBlocks = blocksPerDay * 365 * 2;
+  const reservedToTool = 400000000;
+  const amountReservedToPool = ethers.utils.parseEther(reservedToTool.toString());
+  const minLockTime = 3600 * 24 * 7 * 16;
+
+  const BurnReason = {
+    UnlockMission: 0,
```
Comment (sullof) [inline test/LFGFactoryV2.test.js:39]: This looks like it is not used. Can you clean the file for unused variables?

File: test/LFGFactoryV2.test.js [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+    it("should return the total supply", async function () {
+      expect(await lfg2.totalSupply()).equal("900000000000000000000000000");
+    });
+
+    it("should return the name of the contract", async function () {
+      expect(await lfg2.name()).equal("LFGTokenV2");
+    });
+
+    it("should return the name of the contract", async function () {
```
Comment (sullof) [inline test/LFGFactoryV2.test.js:132]: This is a duplicate.

File: test/LFGFactoryV2.test.js [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+      expect(await lfg2.name()).equal("LFGTokenV2");
+    });
+
+    it("should return the name of the symbol", async function () {
```
Comment (sullof) [inline test/LFGFactoryV2.test.js:136]: Maybe better to call it as "should return the symbol"

File: test/LFGFactoryV2.test.js [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    lfg = await deployUtils.deployProxy(
+      "LFGToken",
+      tokenHolder.address,
+      maxSupply,
+      initialSupply,
+      amountReservedToPool,
+      amountReservedToSharesPool
+    );
+
+    lfg2 = await deployUtils.deploy("LFGTokenV2", "LFGTokenV2", "LFG2", initialSupply, tokenHolder.address);
```
Comment (sullof) [inline test/LFGFactoryV2.test.js:71]: The initial supply of the LFG2 should be the 20% of the maxSupply of the LFG

File: test/LFGFactoryV2.test.js [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+        .to.emit(lfg, "Transfer")
+        .withArgs(addr0, pool.address, amount)
+        .to.emit(pool, "Staked");
+
+      const deposit = await pool.getDeposit(bob.address, 0);
+      const weight = 1 + Math.floor((deposit.lockedUntil - deposit.lockedFrom) / daysInSeconds);
+      let value = deposit.tokenAmount.mul(weight).mul(10).div(100);
+
+      await factory.connect(bob).rewardsFromLfgStakedInCorePool(0);
+      expect(await lfg2.balanceOf(bob.address)).equal(value.toString());
```
Comment (sullof) [inline test/LFGFactoryV2.test.js:221]: We should test more cases where it should revert. For example, if trying to get rewards from someone else's deposit, if the deposit does not exist, etc.

**Review by sullof — CHANGES_REQUESTED** [review]

File: test/LFGFactoryV2.test.js (hunk 5 above)
Comment (roquesrolando) [reply test/LFGFactoryV2.test.js:221]: Yes I was going to test more edge cases I just made this PR so in your PR the functions were correct.
